IDR/production/1.compute_statistics.py

```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+import pathlib, sys
+from tqdm import tqdm
+from utils.statistics import collect_databank_stats, collect_study_stats
+
+# Define path to extraction_utils directory
+parent_dir = str(pathlib.Path(__file__).parents[1])
+sys.path.append(parent_dir)
+from metadata_extraction.extraction_utils.io import walk
+
+
+def nested_dict_pairs_iterator(elements_and_counts_dict):
+    """Collects all combinations of tuples for a nested dictionary
+
+    Parameters
+    ----------
+    elements_and_counts_dict: dict
+        Nested dictionary of all combinations of all unique entries of a metadata file
+        E.g. {study_name: {screen_id: {image_attribute: {unique_element: count}}}}
+
+    Returns
+    -------
+    All possible combinations of tuples for a nested dictionary
+        E.g. (study_name1, screen_id1, image_attribute1, unique_element1, count1)
+    """
+    # Iterate over all key-value pairs of dict argument
+    for key, value in elements_and_counts_dict.items():
+        # Check if value is of dict type
+        if isinstance(value, dict):
+            # If value is dict then iterate over all its values
+            for pair in nested_dict_pairs_iterator(value):
+                yield (key, *pair)
+        else:
+            # If value is not dict type then yield the value
+            yield (key, value)
+
+
+if __name__ == "__main__":
+    # Define study metadata directory
+    studies_metadata_dir = pathlib.Path("IDR/data/metadata")
+
+    # Collect metadata file paths
+    metadata_files = list(walk(studies_metadata_dir))
+    metadata_files = metadata_files[0:5]
```

Comment:
This seems like it should be removed?

Reply:
You are correct! I removed this.

Added in f48ed68c473080a15a6441ad8cee9cd2714b8b24.